File: 9.PalindromeNumber.js
var isPalindrome = function (x) {
  if (x == null || x < 0) {
    return false;
  }

  const inputValue = x;
  let reversed = 0;

  while (x !== 0) {
    reversed = reversed * 10 + x % 10;
    x = parseInt(x / 10);
  }

  return inputValue === reversed;
};

File: 9.PalindromeNumber.java
class Solution {
  public boolean isPalindrome(int x) {
    if (x < 0) {
      return false;
    }

    int inputValue = x;
    int reversed = 0;

    while (x != 0) {
      int temp = reversed * 10 + x % 10;
      if (temp / 10 != reversed) {
        return false;
      }
      reversed = temp;
      x = x / 10;
    }

    return inputValue == reversed;
  }
}